portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -32,20 +37,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String urlToRedirectTo = "/";
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
+      String nickname = getUserNickname(userService.getCurrentUser().getUserId());
       String userEmail = userService.getCurrentUser().getEmail();
       String logoutUrl = userService.createLogoutURL(urlToRedirectTo);
-      response.getWriter().println(createJson("yes", logoutUrl, userEmail));
+      response.getWriter().println(createJson("yes", logoutUrl, userEmail, nickname));
     } else {
       String loginUrl = userService.createLoginURL(urlToRedirectTo);
-      response.getWriter().println(createJson("no", loginUrl, ""));
+      response.getWriter().println(createJson("no", loginUrl, "", ""));
     }
   }
 
-  private String createJson(String isLoggedIn, String url, String email){
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Add some documentation, mention that this is invoked when a user sets a nickname. It took me time to figure it out